src/lib/draw/route/RouteControls.svelte

```diff
@@ -22,6 +22,28 @@
   function toggleSnap() {
     $routeTool!.toggleSnapMode();
   }
+
+  function loadNodes(show: boolean) {
+    if (show && $showAllNodesGj.features.length == 0) {
+      $showAllNodesGj = JSON.parse($routeTool!.inner.debugSnappableNodes());
+    }
+  }
+  $: loadNodes($showAllNodes);
+
+  // Show snappable nodes when going from freehand to snapped
+  let first = true;
+  function freeToSnapped(snapped: boolean) {
```

Comment:
My expectation of this would work is that we would somehow listen to when the first snapped node is placed, when that happens switch `showAllNodes` off.

Also I would expect when `snapped` is false here we'd set `$showAllNodes` to false.

Assuming this is desired behaviour, though, the code looks fine.